modules/nlu/src/backend/pipelines/intents/ft_classifier.ts

```diff
@@ -39,59 +40,118 @@ export default class FastTextClassifier implements IntentClassifier {
     return intents.length > 0 && datasetSize > 0
   }
 
-  async train(intents: sdk.NLU.IntentDefinition[]): Promise<Buffer | undefined> {
-    if (this._hasSufficientData(intents)) {
-      const dataFn = tmp.tmpNameSync({ postfix: '.txt' })
-      await this._writeTrainingSet(intents, dataFn)
+  private async _trainForOneModel(
+    intents: sdk.NLU.IntentDefinition[],
+    modelName: string
+  ): Promise<{ ft: sdk.MLToolkit.FastText.Model; data: Buffer }> {
+    const dataFn = tmp.tmpNameSync({ prefix: modelName + '-', postfix: '.txt' })
+    await this._writeTrainingSet(intents, dataFn)
 
-      const modelFn = tmp.tmpNameSync({ postfix: '.bin' })
+    const modelFn = tmp.tmpNameSync({ postfix: '.bin' })
 
-      // TODO Apply parameters from Grid-search here
-      const ft = new this.toolkit.FastText.Model()
-      await ft.trainToFile('supervised', modelFn, {
-        input: dataFn,
-        loss: 'hs',
-        dim: 15,
-        wordNgrams: 3,
-        minCount: 1,
-        minn: 3,
-        maxn: 6,
-        bucket: 25000,
-        epoch: 50,
-        lr: 0.8
-      })
+    // TODO Apply parameters from Grid-search here
+    const ft = new this.toolkit.FastText.Model()
+    await ft.trainToFile('supervised', modelFn, {
+      input: dataFn,
+      loss: 'hs',
+      dim: 15,
+      wordNgrams: 3,
+      minCount: 1,
+      minn: 3,
+      maxn: 6,
+      bucket: 25000,
+      epoch: 50,
+      lr: 0.8
+    })
+
+    this.model = ft
+
+    return { ft, data: readFileSync(modelFn) }
+  }
 
-      this.model = ft
+  async train(intents: sdk.NLU.IntentDefinition[]): Promise<IntentModel[]> {
+    const contextNames = _.uniq(_.flatMap(intents, x => x.contexts))
 
-      return readFileSync(modelFn)
-    } else {
-      return undefined
+    const models: IntentModel[] = []
+    const modelsByContext: { [key: string]: sdk.MLToolkit.FastText.Model } = {}
+
+    for (const context of contextNames) {
+      // TODO Make the `none` intent undeletable, mandatory and pre-filled with random data
```

Comment:
IMO we should not train the none intent ... it should be handled strictly by the engine which flags none intent no intent passes the confidence threshold.

Reply:
the thing here is if models have only one label, they will always 100% match the 1 label available